### PR TITLE
Add missing headers

### PR DESCRIPTION
**Pull request by marcospereira**
## Purpose

There are a number of files missing the license header. This adds the header for Scala and Java files, including the docs.

### CODE REVIEW

**Comment by mkurz** [comment]
Shouldn't this be checked by the sbt-header plugin?

File: framework/bin/checkFileHeaders [hunk 1]
```diff
@@ -6,4 +6,4 @@
 
 cd $FRAMEWORK
 
-build checkHeaders
+build checkHeaders test:checkHeaders
```
Comment (marcospereira) [inline framework/bin/checkFileHeaders]: @mkurz this changes the ci to check test code too.

Comment (mkurz) [inline framework/bin/checkFileHeaders]: OK, I missed that :see_no_evil:

**Comment by mkurz** [comment]
BTW I am working on upgrading sbt-header to v2.0. However I am waiting to get https://github.com/sbt/sbt-header/issues/117 merged before because otherwise we would have to modify all the headers to append a empty line after them.

**Pull request merged**